File: cpp/examples/task_board/task_filters/task_filters.cpp
#include "task_filters.hpp"

#include <utility>

namespace task_board_example
{

void TaskFilters::setCallbacks(Callback onApply, Callback onReset)
{
    onApply_ = std::move(onApply);
    onReset_ = std::move(onReset);
}

void TaskFilters::onApplyFiltersBtnClick()
{
    if (onApply_)
    {
        onApply_();
    }
}

void TaskFilters::onResetFiltersBtnClick()
{
    if (onReset_)
    {
        onReset_();
    }
}

}  // namespace task_board_example

File: go/examples/task_board/task_filters/task_filters.go
package task_filters

type Callback func()

type TaskFilters struct {
	*TaskFiltersUI
	onApply Callback
	onReset Callback
}

func NewTaskFilters() *TaskFilters {
	filters := &TaskFilters{TaskFiltersUI: NewTaskFiltersUI()}
	filters.SetEventHandler(filters)
	return filters
}

func (filters *TaskFilters) SetCallbacks(onApply Callback, onReset Callback) {
	filters.onApply = onApply
	filters.onReset = onReset
}

func (filters *TaskFilters) OnApplyFiltersBtnClick() {
	if filters.onApply != nil {
		filters.onApply()
	}
}

func (filters *TaskFilters) OnResetFiltersBtnClick() {
	if filters.onReset != nil {
		filters.onReset()
	}
}
